### PR TITLE
 CI/CD: Linting and testing our code before merging a pull request

### PR DESCRIPTION
**Pull request by BRIHMAT-Naoui**
Github Action for linting and testing our code before merging a pull request.

### CODE REVIEW

File: .github/workflows/lint-and-test.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+        node-version: [14.x, 16.x,22.x]
+
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Install dependencies
+      run: npm ci
```
Comment (alamenai) [inline .github/workflows/lint-and-test.yml:22]: Hi @BRIHMAT-Naoui , I guess we don't have used CI integration yet and this command will not work. To install dependencies we run this command:

`npm run install`

File: src/App.tsx
Comment (alamenai) [inline src/App.tsx:1]: I think that these changes are not related to the pull request purpose which is configuring the pipeline.

File: .github/workflows/lint-and-test.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
+name: Lint and Test
+
+on:
```
Comment (alamenai) [inline .github/workflows/lint-and-test.yml:3]: Hi @BRIHMAT-Naoui , I would suggest to test it in another branch before applying it on the main directly, maybe a `dev` branch, what do you think about that ?

**Comment by alamenai** [comment]
The package installation is failed because we don't have installed CI , I mean the command is `npm ci` while it should be `npm install`:

image

File: .github/workflows/lint-and-test.yml (hunk 2 above)
Comment (BRIHMAT-Naoui) [reply .github/workflows/lint-and-test.yml:3]: yes ala you are right

**Comment by alamenai** [comment]
Hi @BRIHMAT-Naoui , please could you change the target branch from `main` to `dev`.

**Comment by alamenai** [comment]
Hi @BRIHMAT-Naoui , the target branch in the pipeline is dev while the targeted branch is main.

**Review by alamenai — APPROVED** [review]
